src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs

```diff
@@ -176,7 +176,10 @@ protected static SearchParameterQueryGeneratorContext VisitSimpleBinary(BinaryOp
 
         protected static SearchParameterQueryGeneratorContext VisitSimpleString(StringExpression expression, SearchParameterQueryGeneratorContext context, StringColumn column, string value)
         {
-            AppendColumnName(context, column, expression);
+            if (expression.StringOperator != StringOperator.LeftSideStartsWith)
+            {
+                AppendColumnName(context, column, expression);
+            }
```

Comment:
To verify my understanding, in this method we are looking to append the following SQL code to the query:

```
AND @p1 LIKE Uri+'%'
AND Uri NOT  LIKE @p2
```
and, in the case of the :above modifier, we need to specify `@p1 LIKE ` before the column name `Uri` (which is different than the other cases). Is this correct?

Reply:
yes that's the query we expect. I don't know much about second AND where we check it doesn't start with `urn:`, there probably a reason for that.

if `:below` is a check what string 'a' is a prefix of 'b', `:above` is switching `a` and `b` places. 

Another way how I could formulate that expression is by using `CHARINDEX(@p1, Uri)=1`, which would check what @p1 value is prefix of value in Uri column. But as I found you can use LIKE expression as well.